clients/rospy/src/rospy/impl/tcpros_service.py

```diff
@@ -145,15 +146,16 @@ def contact_service(resolved_name, timeout=10.0):
         while not rospy.core.is_shutdown():
             try:
                 if contact_service(resolved_name):
+                    if contact_failed:
+                        rospy.core.loginfo("wait_for_service(%s): finally were able to contact [%s]"%(resolved_name, uri))
                     return
             except KeyboardInterrupt:
                 # re-raise
                 rospy.core.logdebug("wait_for_service: received keyboard interrupt, assuming signals disabled and re-raising")
                 raise 
             except: # service not actually up
-                if first:
-                    first = False
-                    rospy.core.logerr("wait_for_service(%s): failed to contact [%s], will keep trying"%(resolved_name, uri))
+                contact_failed = True
+                rospy.core.logwarn("wait_for_service(%s): failed to contact [%s], will keep trying"%(resolved_name, uri))
```

Comment:
With #1091 being merged this could be updated to call `rospy.core.logwarn_throttle(10, "...")` to reduce the verbosity of this line.

Same for the other warning above.